packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1466,6 +1470,7 @@ export class ContainerRuntime
 			compressionAlgorithm: CompressionAlgorithms.lz4,
 		};
 
+		assert(isIDeltaManagerInternal(deltaManager), "Invalid delta manager");
```

Comment:
This pattern is Ok to hide internals, but it results in runtime failures in case internal interface is not implemented. It's harder to see contracts.

Would it be better for IContainerContext to have IDeltaManager & IDeltaMaagerInternal?
This exposes internals, but only on that one interface. 

IContainerContext is marked as legacy & alpha. It will always be some form of "public" API, so there is some risk of exposing internals, but I think it's fine to acknowledge that whoever implements instantiateRuntime has access to internals, but our stock implementation (ContainerRuntime, IContainerRuntime(Base)) will not expose internals to users. Same for Container & IContainer - they will not expose internals.

Reply:
Didn't want to expose the delta manager on the container context, since technically it will still be exposed. With this approach we are not exposing it anywhere. 
The container runtime implementor does not need to implement the IDeltaManagerInternal, our Deltamanager class implement both and if the runtime does not need the inbound/outbound queues, then they don't need to do any type checking and if they do, then they can look that the Delta manager implements it and they can assert that like we do here as an example. 
Not sure, if got you completely there.

Reply:
Take a look at IDeltaManagerErased, toDeltaManagerErased() & toDeltaManagerInternal()  and its usage.
We could expose IDeltaManagerInternalErased that way.

For example, IFluidDataStoreRuntime exposed IDeltaManagerErased.

That said, current pattern is fine too, and I'd probably rename toDeltaManagerInternal -> toDeltaManager in any case, to remove confusion on "internal" meaning.